app/models/candidate_pool_application.rb

```diff
@@ -12,6 +13,15 @@ def self.filtered_application_forms(filters)
     ApplicationForm.where(id: scope.select(:application_form_id))
   end
 
+  def self.remove_application_forms_rejected_by_providers(scope, provider_user)
+    return scope if provider_user.blank?
+
+    provider_ids = provider_user.providers.ids
+    rejected_application_ids = where('rejected_provider_ids @> ARRAY[?]::bigint[]', Array.wrap(provider_ids)).pluck(:application_form_id)
```

Comment:
```suggestion
    rejected_application_ids = where('rejected_provider_ids @> ARRAY[?]::bigint[]', Array.wrap(provider_ids)).select(:application_form_id)
```
Using a select here will avoid performing a query and allow it to be optimised by Rails 👌